### ISSUE TITLE
fails to build from source with gcc 14

### ISSUE THREAD

**Issue by linuxmaniac** [SIPp/sipp#760]
From https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1081521

```
Source: sip-tester
Version: 1:3.7.3-1
Severity: serious
Tags: ftbfs

https://buildd.debian.org/status/logs.php?pkg=sip-tester&ver=1%3A3.7.3-1

...
/<<PKGBUILDDIR>>/src/call.cpp: In member function ‘char* call::get_last_header(const char*)’:
/<<PKGBUILDDIR>>/src/call.cpp:1567:32: error: ‘:’ directive writing 1 byte into a region of size between 0 and 2049 [-Werror=format-overflow=]
 1567 |         sprintf(with_colon, "%s:", name);
      |                                ^
In file included from /usr/include/stdio.h:970,
                 from /usr/include/c++/14/cstdio:42,
                 from /usr/include/c++/14/ext/string_conversions.h:45,
                 from /usr/include/c++/14/bits/basic_string.h:4154,
                 from /usr/include/c++/14/string:54,
                 from /usr/include/c++/14/bits/locale_classes.h:40,
                 from /usr/include/c++/14/bits/ios_base.h:41,
                 from /usr/include/c++/14/ios:44,
                 from /usr/include/c++/14/istream:40,
                 from /usr/include/c++/14/fstream:40,
                 from /<<PKGBUILDDIR>>/src/call.cpp:43:
In function ‘int sprintf(char*, const char*, ...)’,
    inlined from ‘char* call::get_last_header(const char*)’ at /<<PKGBUILDDIR>>/src/call.cpp:1567:16:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: note: ‘__builtin___sprintf_chk’ output between 2 and 2051 bytes into a destination of size 2049
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
/<<PKGBUILDDIR>>/src/call.cpp: In member function ‘char* call::get_last_request_uri()’:
/<<PKGBUILDDIR>>/src/call.cpp:1608:16: error: ‘char* __builtin_strncpy(char*, const char*, long unsigned int)’ specified bound depends on the length of the source argument [-Werror=stringop-truncation]
 1608 |         strncpy(last_request_uri, tmp, tmp_len);
      |                ^
/<<PKGBUILDDIR>>/src/call.cpp:1597:21: note: length computed here
 1597 |     tmp_len = strlen(tmp) - strlen(tmp2);
      |               ~~~~~~^~~~~
...
```

**Comment by lemenkov** [comment]
Fixed  in fdc0c97e1d8acbe4de0d89cf26dfe74bf9b413ed.

**Comment by lemenkov** [comment]
Ahh snap! Still a valid complain.

**Comment by orgads** [comment]
Interesting... I couldn't reproduce with `alpine:edge`. Checking now with `debian:unstable`.

**Comment by orgads** [comment]
@lemenkov Did you reproduce this? How?

**Comment by orgads** [comment]
I checked with `debian:unstable` on both x86_64 and i386, and everything compiles fine on latest master.

**Comment by lemenkov** [comment]
> @lemenkov Did you reproduce this? How?

Sorry it was indeed fixed in fdc0c97e1d8acbe4de0d89cf26dfe74bf9b413ed. I accidentally used an old 3.7.3  tag version.

**Comment by linuxmaniac** [comment]
Thanks!
